payjoin/src/ohttp.rs

```diff
@@ -52,6 +52,77 @@ pub fn ohttp_encapsulate(
     Ok((buffer, ohttp_ctx))
 }
 
+#[derive(Debug)]
+pub enum DirectoryResponseError {
+    InvalidSize(usize),
+    OhttpDecapsulation(OhttpEncapsulationError),
+    UnexpectedStatusCode(http::StatusCode),
+}
+
+impl fmt::Display for DirectoryResponseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use DirectoryResponseError::*;
+
+        match self {
+            OhttpDecapsulation(e) => write!(f, "OHTTP Decapsulation Error: {e}"),
+            InvalidSize(size) => write!(
+                f,
+                "Unexpected response size {}, expected {} bytes",
+                size,
+                crate::directory::ENCAPSULATED_MESSAGE_BYTES
+            ),
+            UnexpectedStatusCode(status) => write!(f, "Unexpected status code: {status}"),
+        }
+    }
+}
+
+impl error::Error for DirectoryResponseError {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+        use DirectoryResponseError::*;
+
+        match self {
+            OhttpDecapsulation(e) => Some(e),
+            InvalidSize(_) => None,
+            UnexpectedStatusCode(_) => None,
+        }
+    }
+}
+
+pub fn process_get_res(
+    res: &[u8],
+    ohttp_context: ohttp::ClientResponse,
+) -> Result<Option<Vec<u8>>, DirectoryResponseError> {
+    let response = process_ohttp_res(res, ohttp_context)?;
+    match response.status() {
+        http::StatusCode::OK => Ok(Some(response.body().to_vec())),
+        http::StatusCode::ACCEPTED => Ok(None),
+        _ => Err(DirectoryResponseError::UnexpectedStatusCode(response.status())),
+    }
+}
+
+pub fn process_post_res(
+    res: &[u8],
+    ohttp_context: ohttp::ClientResponse,
+) -> Result<(), DirectoryResponseError> {
+    let response = process_ohttp_res(res, ohttp_context)?;
+    match response.status() {
+        http::StatusCode::OK => Ok(()),
+        _ => Err(DirectoryResponseError::UnexpectedStatusCode(response.status())),
+    }
```

Comment:
I think this is the idiom
```suggestion
    match response.status() {
        http::StatusCode::OK => Ok(()),
        status_code => Err(DirectoryResponseError::UnexpectedStatusCode(status_code)),
    }
```